modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -1338,6 +1343,50 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
     )
   }
 
+  private def renderValidatedNewtype(
+      shapeId: ShapeId,
+      name: NameRef,
+      tpe: Type,
+      recursive: Boolean,
+      hints: List[Hint]
+  ): Lines = {
+    val validator = {
+      val tags = hints.collect { case t: Hint.Constraint => t }
+      tags match {
+        case h :: tail =>
+          (
+            line".validating(${renderNativeHint(h.native)})" +:
+              tail.map { tag => line".alsoValidating(${renderNativeHint(tag.native)})" }
+          ).intercalate(Line.empty)
+        case _ => Line.empty
+      }
+    }
+
+    val definition =
+      if (recursive) line"$recursive_("
+      else Line.empty
+    val trailingCalls =
+      line".withId(id).addHints(hints)${renderConstraintValidation(hints)}"
+    val closing = if (recursive) ")" else ""
+    lines(
+      documentationAnnotation(hints),
+      deprecationAnnotation(hints),
+      obj(name, line"$NewtypeValidated_[$tpe]")(
+        renderId(shapeId),
+        renderHintsVal(hints),
+        line"val underlyingSchema: $Schema_[$tpe] = ${tpe.schemaRef}$trailingCalls",
+        lines(
+          line"implicit val schema: $Schema_[$name] = $definition$bijection_(underlyingSchema, asBijectionUnsafe)$closing"
```

Comment:
the schema of `NewtypeValidated` should not be a bijection, it should be a refinement

Reply:
To be more precise : the schema should not call upon `asBijectionUnsafe`. 

I think the easiest way to achieve this would be to add a `def toSchema(a: Schema[A]) : Schema[B]` to the `Validator[A, B]` interface, because the `ValidatorImpl` keeps everything that you'll need to formulate the correct schema : a list of refinements and a bijection to apply at the end.

Reply:
Just to make sure that I understand what the expected the implementation of `toSchema` to be. I can think of something like this:

```scala
override def toSchema(a: Schema[A]): Schema[B] = 
  refinements.foldLeft(a) { (schema, refinement) => 
    schema.refined[B](refinement.constraint)
  }.biject(bijection) // problem here is that the bijection is still of the "unsafe kind"
```

and it would be invoked like so:

```scala
// ValidatedString

val underlyingSchema: Schema[String] = string.withId(id).addHints(hints).validated(smithy.api.Length(min = Some(1L), max = None))

val validator: Validator[String, ValidatedString] = Validator.of[String, ValidatedString].validating(smithy.api.Length(min = Some(1L), max = None))

val schema: Schema[ValidatedString] = validator.toSchema(string.withId(id).addHints(hints))
```

Is that close to what you had in mind?

Reply:
Sorry, I've been quite busy and had forgotten to reply to this.

Yes, that's pretty much what I had in mind 👍

Reply:
Done